src/main/cpp/src/NativeParquetJni.cpp

```diff
@@ -106,215 +125,272 @@ class column_pruner {
      */
     column_pruner(const std::vector<std::string> & names, 
             const std::vector<int> & num_children, 
-            int parent_num_children): children(), s_id(0), c_id(-1) {
-      add_depth_first(names, num_children, parent_num_children);
+            const std::vector<int> & tags, 
```

Comment:
I'm not sure about the code format in this repo. If we try to follow rapids, we should use clangformat config from cudf. If we have our own standard then we should also define our own clangformat config.